tools/c7n_azure/c7n_azure/resources/cosmos_db.py

```diff
@@ -278,9 +312,76 @@ def process_resource_set(self, resources):
 
         return matched
 
-    @staticmethod
-    def account_key(resource):
-        return resource['c7n:document-endpoint']
+
+@CosmosDBCollection.action_registry.register('replace-offer')
+class CosmosDBReplaceOfferAction(AzureBaseAction):
+    """CosmosDB Replace Offer Action
+
+    Modify the throughput of a cosmodb collection's offer
+
+    :example:
+
+    This policy will ensure that no collections have offers with more than 400 RU/s throughput.
+
+    .. code-block:: yaml
+
+        policies:
+          - name: limit-throughput-to-400
+            resource: azure.cosmosdb-collection
+            filters:
+              - type: offer
+                key: content.offerThroughput
+                op: gt
+                value: 400
+            actions:
+              - type: replace-offer
+                throughput: 400
+
+    """
+
+    schema = type_schema(
+        'replace-offer',
+        required=['throughput'],
+        **{
+            'throughput': {'type': 'number'}
```

Comment:
If you want you can enforce >0 (and even maximum, if there is one) here.  `'throughput': {'type': 'number', 'minimum': 0}`